MIN is used but not defined in three sources using it

## Background information
I am trying to compile a cloned version of OpenMPI including PRRTE under Solaris 11.4 where I get compilation errors.
### What version of the PMIx Reference RTE (PRRTE) are you using? (e.g., v2.0, v3.0, git master @ hash, etc.)
Commit https://github.com/openpmix/prrte/commit/e0076a116884c9a8bcd5fffe20be84f3564446c3
##### What version of PMIx are you using? (e.g., v4.2.0, git branch name and hash, etc.)
Commit https://github.com/openpmix/openpmix/commit/5bd97056ccd994c711fde4ac19aded8e4eac8285
### Please describe the system on which you are running

* Operating system/version: Solaris 11.4
* Compiler: gcc 11.2.0
* Computer hardware:  64 bit x86 architecture
* Network type: not relevant here

-----------------------------

## Details of the problem

MIN is used in multiple sources but remains undefined, leading to following building error:

```
libtool: link: /opt/ulm/dublin/cmd/gcc -DPRTE_CONFIGURE_USER=\"root\" -DPRTE_CONFIGURE_HOST=\"theon\" "-DPRTE_CONFIGURE_DATE=\"Wed Jan 25 10:23:18 UTC 2023\"" -DPRTE_BUILD_USER=\"root\" -DPRTE_BUILD_HOST=\"theon\" "-DPRTE_BUILD_DATE=\"Wed Jan 25 15:42:11 UTC 2023\"" "-DPRTE_BUILD_CFLAGS=\"-O3 -DNDEBUG  -D_REENTRANT -finline-functions \"" "-DPRTE_BUILD_CPPFLAGS=\"-iquote/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/prrte -iquote/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/prrte/src/include  -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/hwloc-2.7.1/include -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/hwloc-2.7.1/include -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix/include -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix/include -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix/ -I/home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix/ -D_REENTRANT \"" "-DPRTE_BUILD_LDFLAGS=\"  \"" "-DPRTE_BUILD_LIBS=\"-lm   -levent_core -levent_pthreads /home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix/src/libpmix.la /home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/hwloc-2.7.1/hwloc/libhwloc.la\"" -DPRTE_CC_ABSOLUTE=\"\" -DPRTE_GREEK_VERSION=\"a1\" -DPRTE_REPO_REV=\"psrvr-v2.0.0rc1-4537-ge0076a1168\" -DPMIX_RELEASE_DATE=\"@PMIX_RELEASE_DATE@\" -O3 -DNDEBUG -D_REENTRANT -finline-functions -o .libs/prted prted.o  ../../../src/.libs/libprrte.so /home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/openpmix/src/.libs/libpmix.so -ldl -levent_core -levent_pthreads /home/pkgdev/dublin/openmpi/src/ompi-master/3rd-party/hwloc-2.7.1/hwloc/.libs/libhwloc.so -lm -lpicl -lkstat -llgrp -Wl,-rpath -Wl,/opt/ulm/dublin/lib/amd64
/opt/ulm/dublin/bin/ld: ../../../src/.libs/libprrte.so: undefined reference to `MIN'
collect2: error: ld returned 1 exit status
```

MIN is used in a number of sources:

```
theon$ find . -name '*.c' | xargs grep -l '\<MIN('
./mca/oob/tcp/oob_tcp_component.c
./mca/prtereachable/weighted/reachable_weighted.c
./mca/prtereachable/netlink/reachable_netlink_module.c
./util/bipartite_graph.c
theon$ 
```

But it is defined only within `util/bipartite_graph.c`:

```c
#ifndef MIN
#    define MIN(a, b) ((a) < (b) ? (a) : (b))
#endif
```

Hence, it is missing in the other sources listed above.

I would suggest to replace the MIN & MAX macros in `bipartite_graph.c` by some inline function that is defined within the header files and include it the listed C sources.

--- a/src/mca/oob/tcp/oob_tcp_component.c
+++ b/src/mca/oob/tcp/oob_tcp_component.c
@@ -21,7 +21,7 @@
  * Copyright (c) 2017      IBM Corporation.  All rights reserved.
  * Copyright (c) 2020      Amazon.com, Inc. or its affiliates.  All Rights
  *                         reserved.
- * Copyright (c) 2021-2022 Nanook Consulting.  All rights reserved.
+ * Copyright (c) 2021-2023 Nanook Consulting.  All rights reserved.
  * $COPYRIGHT$
  *
  * Additional copyrights may follow
@@ -59,6 +59,10 @@
 #include <arpa/inet.h>
 #include <ctype.h>
 #include <sys/socket.h>
+
+#ifndef MIN
+#    define MIN(a, b) ((a) < (b) ? (a) : (b))
+#endif
 
 #include "src/class/pmix_list.h"
 #include "src/event/event-internal.h"

--- a/src/mca/prtereachable/netlink/reachable_netlink_module.c
+++ b/src/mca/prtereachable/netlink/reachable_netlink_module.c
@@ -4,7 +4,7 @@
  * Copyright (c) 2015-2020 Cisco Systems, Inc.  All rights reserved
  * Copyright (c) 2017 Amazon.com, Inc. or its affiliates.
  *                    All Rights reserved.
- * Copyright (c) 2021-2022 Nanook Consulting.  All rights reserved.
+ * Copyright (c) 2021-2023 Nanook Consulting.  All rights reserved.
  * $COPYRIGHT$
  *
  * Additional copyrights may follow
@@ -18,6 +18,10 @@
 
 #ifdef HAVE_MATH_H
 #    include <math.h>
+#endif
+
+#ifndef MIN
+#    define MIN(a, b) ((a) < (b) ? (a) : (b))
 #endif
 
 #include "libnl_utils.h"

--- a/src/mca/prtereachable/weighted/reachable_weighted.c
+++ b/src/mca/prtereachable/weighted/reachable_weighted.c
@@ -8,7 +8,7 @@
  * Copyright (c) 2017      Amazon.com, Inc. or its affiliates.
  *                         All Rights reserved.
  * Copyright (c) 2020      Cisco Systems, Inc.  All rights reserved
- * Copyright (c) 2021-2022 Nanook Consulting.  All rights reserved.
+ * Copyright (c) 2021-2023 Nanook Consulting.  All rights reserved.
  * $COPYRIGHT$
  *
  * Additional copyrights may follow
@@ -26,6 +26,10 @@
 #endif
 #ifdef HAVE_MATH_H
 #    include <math.h>
+#endif
+
+#ifndef MIN
+#    define MIN(a, b) ((a) < (b) ? (a) : (b))
 #endif
 
 #include "reachable_weighted.h"

